src/cls/timeindex/cls_timeindex.cc

```diff
@@ -155,6 +154,7 @@ static int cls_timeindex_list(cls_method_context_t hctx,
               index.c_str(), e.key_ext.c_str(), bl.length());
       e.value = bl;
       entries.push_back(e);
+      marker = index;
```

Comment:
it's probably a good idea to advance the marker if `parse_index()` fails, too. there's no point in revisiting that marker if the parse will fail next time. so i think this line should move down, outside of this `else{}` block - do you buy that?

Reply:
Thanks for the review @cbodley. I've thought about it, but doing so would trim away the hint for that index. Would it be fine to trim away entries leaving behind objects? The side effect of those orphaned objects are that, one cannot `swift stat` them as they fail at the precondition check for expired epoch and one cannot find their expiration hint present in any of the shards either. One can delete them manually (though the deletes throw a spurious 404!) because of this check https://github.com/ceph/ceph/blob/master/src/rgw/rgw_op.cc#L4250

Reply:
presumably, `parse_index()` would only fail if there was a bug on insert.. but if it _does_ fail here, that entry is omitted from the results - it's effectively invisible to the client forever. so there's nothing the client could do to recover, whether we trim it or not. advancing the marker in that case is a (very small) optimization because we won't have to revisit that entry in the next iteration of `RGWObjectExpirer::process_single_shard()`